include/SampleTrack.h

```diff
@@ -123,10 +130,14 @@ public slots:
 private:
 	SampleTCO * m_tco;
 
-	SampleBufferVisualizer m_sampleBufferVisualizer;
-} ;
+	std::shared_ptr<std::mutex> m_sampleBufferVisualizerMutex = std::make_shared<std::mutex>();
```

Comment:
`std::mutex` is unavailable in MinGW with win32 threading model as well as `std::thread`. You can use [mingw-std-threads](https://github.com/meganz/mingw-std-threads) as in 48cdce9a790b6ba7384a36d5b4a6ffadc21a89bd or switch to `QMutex`.
Also, don't forget to include `<mutex>` if you're going to stick to `std::mutex`. MSVC complains about it.

Reply:
Thread sanitizer doesn't work well with QMutex. However, I'll just stick to it for now and then add that option.